app/src/main/java/com/humberto/tasky/auth/presentation/registration/RegisterScreen.kt

```diff
@@ -0,0 +1,254 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.humberto.tasky.R
+import com.humberto.tasky.auth.domain.UserDataValidator
+import com.humberto.tasky.core.presentation.designsystem.CheckIcon
+import com.humberto.tasky.core.presentation.designsystem.CrossIcon
+import com.humberto.tasky.core.presentation.designsystem.TaskyError
+import com.humberto.tasky.core.presentation.designsystem.TaskyGreen
+import com.humberto.tasky.core.presentation.designsystem.TaskyTheme
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyActionButton
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyFloatingActionButton
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyPasswordTextField
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyScaffold
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyTextField
+import com.humberto.tasky.core.presentation.ui.ObserveAsEvents
+
+@Composable
+fun RegisterScreenRoot(
+    onRegisterSuccess: () -> Unit,
+    onBackClick: () -> Unit,
+    viewModel: RegisterViewModel = hiltViewModel()
+) {
+    val context = LocalContext.current
+    val keyboardController = LocalSoftwareKeyboardController.current
+    ObserveAsEvents(viewModel.events) { event ->
+        when(event) {
+            is RegisterEvent.Error -> {
+                keyboardController?.hide()
+                Toast.makeText(
+                    context,
+                    event.error.asString(context),
+                    Toast.LENGTH_LONG
+                ).show()
+            }
+            RegisterEvent.RegisterSuccess -> {
+                keyboardController?.hide()
+                Toast.makeText(
+                    context,
+                    R.string.login_successful,
+                    Toast.LENGTH_LONG
+                ).show()
+
+                onRegisterSuccess()
+            }
+        }
+    }
+    RegisterScreen(
+        state = viewModel.state,
+        onAction = { action ->
+            when(action) {
+                is RegisterAction.OnBackClick -> onBackClick()
+                else -> Unit
+            }
+            viewModel.onAction(action)
+        }
+    )
+}
+
+@Composable
+private fun RegisterScreen(
+    state: RegisterState,
+    onAction: (RegisterAction) -> Unit
+) {
+    val (
+        nameFocusRequester,
+        emailFocusRequester,
+        passwordFocusRequester,
+    ) = FocusRequester.createRefs()
```

Comment:
If I'm not wrong this call should be inside a `remember` block, otherwise it would create new refs on each recomposition